genesis/vis/camera.py

```diff
@@ -505,13 +492,23 @@ def render_pointcloud(self, world_frame=True):
         """
         # Compute the (denormalized) depth map using PyRender systematically.
         # TODO: Add support of BatchRendered (requires access to projection matrix)
-        self._rasterizer.update_scene()
-        rgb_arr, depth_arr, seg_idxc_arr, normal_arr = self._rasterizer.render_camera(
-            self, rgb=False, depth=True, segmentation=False, normal=False
-        )
+        if self._batch_renderer is not None:
+            if self._batch_renderer.use_rasterizer:
+                _, depth_arr, _, _ = self._batch_render(
+                    rgb=False, depth=True, segmentation=False, normal=False,
+                )
+            else:
+                gs.raise_exception("point cloud only support in BatchRender with rasterizer.")
+        elif self._rasterizer is not None:
+            self._rasterizer.update_scene()
+            _, depth_arr, _, _ = self._rasterizer.render_camera(
+                self, rgb=False, depth=True, segmentation=False, normal=False
+            )
+        else:
+            gs.raise_exception("Render point cloud not supported.")
 
         # Convert OpenGL projection matrix to camera intrinsics
-        P = self._rasterizer._camera_nodes[self.uid].camera.get_projection_matrix()
+        P = self.projection_matrix
```

Comment:
Since you are computing the projection matrix manually, you could just extract `fx, fy, cx, cy` directly from camera parameters without going through the projection matrix. Could you change this?

Yet, do not remove the projection matrix property, it is nice to have it anyway.